pydantic/color.py

```diff
@@ -246,20 +247,20 @@ def parse_str(value: str) -> RGBA:
             alpha = None
         return ints_to_rgba(r, g, b, alpha)
 
-    m = r_rgb.fullmatch(value_lower)
+    m = re.fullmatch(r_rgb, value_lower)
     if m:
         return ints_to_rgba(*m.groups(), None)  # type: ignore
 
-    m = r_rgba.fullmatch(value_lower)
+    m = re.fullmatch(r_rgba, value_lower)
     if m:
         return ints_to_rgba(*m.groups())  # type: ignore
 
-    m = r_hsl.fullmatch(value_lower)
+    m = re.fullmatch(r_hsl, value_lower)
```

Comment:
Wait, isn't use of `re.fullmatch` instead of using compiled patterns slowed down function calls?

Reply:
only very small, `re.fullmatch` keeps a cache of compiled regexes to use after the first call.

So the old slow down is looking up the cache of regexes.